discord/interactions.py

```diff
@@ -188,6 +190,12 @@ def _from_data(self, data: InteractionPayload):
         self.user: User | Member | None = None
         self._permissions: int = 0
 
+        self._guild: Guild | None = None
+        if self.guild is None and (_guild_data := data.get("guild")):
+            self._guild = Guild(data=data, state=self)
+
+        self._guild_data = _guild_data
```

Comment:
Simpler? (also corrected the data argument)
```suggestion
        self._guild_data = data.get("guild")
        if self.guild is None and self._guild_data:
            self._guild = Guild(data=self._guild_data, state=self)
```